pkg/config/remote/client.go

```diff
@@ -8,51 +8,47 @@ import (
 
 	"github.com/DataDog/datadog-agent/pkg/api/security"
 	"github.com/DataDog/datadog-agent/pkg/config/remote/data"
-	"github.com/DataDog/datadog-agent/pkg/config/remote/uptane"
+	"github.com/DataDog/datadog-agent/pkg/config/remote/meta"
 	"github.com/DataDog/datadog-agent/pkg/proto/pbgo"
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/client"
 	agentgrpc "github.com/DataDog/datadog-agent/pkg/util/grpc"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/version"
 	"github.com/pkg/errors"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/metadata"
 )
 
 // Client is a remote-configuration client to obtain configurations from the local API
 type Client struct {
-	m               sync.Mutex
-	ctx             context.Context
-	close           func()
-	facts           Facts
-	enabledProducts map[data.Product]struct{}
-	pollInterval    time.Duration
+	m     sync.Mutex
+	ctx   context.Context
+	close func()
 
-	grpc          pbgo.AgentSecureClient
-	partialClient *uptane.PartialClient
-	configs       *configs
+	agentName    string
+	products     []string
+	pollInterval time.Duration
 
-	lastPollErr error
+	grpc           pbgo.AgentSecureClient
+	stateClient    *client.Client
```

Comment:
We have a Client here in `pkg/config/remote` and a Client in `pkg/remoteconfig`. Is there anything we can do to make it more clear how they differ? To be honest, I'm drawing a blank - but I can definitely see this being a source of confusion for newcomers down the road. It's not even something we need to address immediately, but it's something worth brainstorming about probably.

Reply:
Yeah I agree it's not super great currently, the next step is going to be moving `pkg/remoteconfig/client` to the universal client repository and importing it from here. We can improve the name (`remoteconfig/state` or something ?) at that point.